tools/tplgtool2.py

```diff
@@ -882,14 +882,23 @@ def arg_split(value: str) -> "list[str]":
         "Split comma-delimited arguements value."
         return [v.strip() for v in value.split(',')]
 
+    def glob_or_abs(rootpath: Path, filename: str) -> Path:
+        # this is a workround for Path.glob(): it didn't implement for absolute path,
+        # even though the document said the path can be absolute:
+        # https://docs.python.org/3.8/library/glob.html#glob.glob
+        if os.path.isabs(filename):
+            rootpath = Path('/')
+            filename = os.path.relpath(filename, '/')
+        return rootpath.glob(filename)
+
     def get_tplg_paths(tplgroot: typing.Union[str, None], filenames: str) -> "list[Path]":
         rootpath = Path.cwd() if tplgroot is None else Path(tplgroot.strip())
         if not rootpath.is_dir():
             raise NotADirectoryError(rootpath)
         return [
             fullpath 
             for filename in arg_split(filenames) 
-            for fullpath in rootpath.glob(filename) 
+            for fullpath in glob_or_abs(rootpath, filename) 
```

Comment:
It finally makes sense but I found the 2-level list comprehension hard to follow. Returning a generator called `glob_or_abs` is not very intuitive either. The "or_abs" is an internal implementation detail, users of that function don't need to know about it.


Maybe this:
```python
return [  ​topology
   ​for pattern in patterns.split(',')
   for topology in do_glob(basepath, pattern)
]
```